repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java

```diff
@@ -446,7 +446,8 @@ private void addTagPropagation(AtlasVertex fromVertex, AtlasVertex toVertex, Atl
     }
 
     public List<AtlasVertex> addTagPropagation(AtlasVertex classificationVertex, List<AtlasVertex> propagatedEntityVertices) throws AtlasBaseException {
-        List<AtlasVertex> ret = null;
+        AtlasPerfMetrics.MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("addTagPropagation");
+        List<AtlasVertex> ret = new ArrayList<>();
```

Comment:
i think its better if we don't change the allocation logic, in the old way we only create the new arraylist when we want to add

Reply:
but not important if addTagPropagation is not called very frequently

Reply:
Actually, if we were considering the heap usage, the size of an empty list is zero.